jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/TargetSystemIntegrationTest.java

```diff
@@ -125,6 +125,8 @@ void endToEndTest(@TempDir Path tmpDir) {
     verifyMetrics();
   }
 
+  // TODO: This implementation is DEPRECATED and will be removed once all integration tests are
+  // migrated to MetricsVerifier
   protected void waitAndAssertMetrics(Iterable<Consumer<Metric>> assertions) {
```

Comment:
[for reviewer] we can't use `@Deprecated` here without introducing compiler warnings and having to use `@SuppressWarnings` in test code, so migrating away from this will be done in multiple steps as part of https://github.com/open-telemetry/opentelemetry-java-contrib/issues/1573